### PR TITLE
Sign in new

### PR DESCRIPTION
**Pull request by Steven-Klavins**

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,3 +1,20 @@
...
+end
+
+def logged_in?  
+  !current_user.nil?  
+end
+
+def authorized
+  redirect_to '/welcome' unless logged_in?
+end
+
```
Comment (hturnbull93) [inline app/controllers/application_controller.rb:19]: Can we sort indentation for this section please :smile:

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  def new
+     @user = User.new
+  end
+  def create
+     @user = User.create(params.require(:user).permit(:username,        
+     :password, :email, :firstname, :lastname))
+     session[:user_id] = @user.id
+     redirect_to '/welcome'
+  end
+  end
```
Comment (hturnbull93) [inline app/controllers/users_controller.rb:14]: indentation here please

File: app/views/sessions/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<h1>Login</h1>
+   <%= form_tag '/login' do %>
+   <%= label_tag :email%>
+   <%= text_field_tag :email %>
+   <%= label_tag :password%>
+   <%= password_field_tag :password%>
+   <%= submit_tag "Login"%>
+<%end%>
```
Comment (hturnbull93) [inline app/views/sessions/new.html.erb:8]: indentation here please

File: config/routes.rb [hunk 4]
```diff
@@ -1,5 +1,11 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  resources :users, only: [:new, :create]
 
-  resources :posts
```
Comment (hturnbull93) [inline config/routes.rb:4]: This will be needed for posts to be accessed I think. When running a dev erver I cannot access `/posts`

Comment (Edward-Phillips) [inline config/routes.rb:4]: It looks like this removes the posts functionality, is there a way you can add users without removing the posts resource?

**Review by Edward-Phillips — CHANGES_REQUESTED** [review]
Nice, It's great being able to sign in and have that stored nicely - it does seem like the posts functionality may have been damaged by this though, it looks like it's come from where the routes.rb file has changed, other than that it looks great, maybe would've been nice to have a feature test for signing up and signing in, but it works!

**Review by hturnbull93 — CHANGES_REQUESTED** [review]
The main functionality of signing up and getting into a session looks good! 
However for some reason, not sure what that is the posts functionality is now missing?
There are no feature tests I can see, so it is unclear precisely what what process you guys used to create this.
Manual testing also reveals that the emails are not required to be a unique column in the database, so I was able to sign up separately as two users, with the same email address. (this might be a separate ticket though, not sure)
Screenshot from 2020-05-06 16-10-00

**Pull request closed without merging**